django/views/generic/edit.py

```diff
@@ -221,17 +224,30 @@ def get_success_url(self):
         if self.success_url:
             return self.success_url.format(**self.object.__dict__)
         else:
-            raise ImproperlyConfigured(
-                "No URL to redirect to. Provide a success_url.")
+            raise ImproperlyConfigured("No URL to redirect to. Provide a success_url.")
 
 
-class BaseDeleteView(DeletionMixin, BaseDetailView):
+class BaseDeleteView(DeletionMixin, FormMixin, SingleObjectMixin, ProcessFormView):
     """
     Base view for deleting an object.
 
     Using this base class requires subclassing to provide a response mixin.
     """
 
+    form_class = Form
+
+    def get(self, request, *args, **kwargs):
+        self.object = self.get_object()
+        return super().get(request, *args, **kwargs)
+
+    def delete(self, request, *args, **kwargs):
+        self.object = self.get_object()
+        # Process form instead of directly deleting the object.
+        return super(DeletionMixin, self).post(request, *args, **kwargs)
```

Comment:
I don't think we should touch the delete() behaviour here — if you're sending a `DELETE` HTTP request, you're not going to want us to suddenly change that under you. 

I think that means adjusting the inheritance ordering and adding a `post()` to set `self.object`.

Reply:
Agreed. Do we need to adjust the inheritance? Please check the new commit.